collection-scripts/gather_network_trace

```diff
@@ -0,0 +1,146 @@
+
+    oc -n openshift-ovn-kubernetes logs ${OVS_POD} 2>/dev/null | \
+      convert_ovs_timestamp "${NODE}" "${OVS_POD}" "ovs-daemons" \
+      > ${NETWORK_TRACE_PATH}/${NODE}_${OVS_POD}_ovs-daemons.log &
+    PIDS+=($!)
+
+    oc -n openshift-ovn-kubernetes logs ${OVS_POD} -p 2>/dev/null | \
+      convert_ovs_timestamp "${NODE}" "${OVS_POD}" "ovs-daemons" \
+      > ${NETWORK_TRACE_PATH}/${NODE}_${OVS_POD}_ovs-daemons-previous.log &
+    PIDS+=($!)
+
+    oc -n openshift-ovn-kubernetes logs ${OVNKUBE_NODE_POD} -c ovn-controller 2>/dev/null | \
+      convert_ovs_timestamp "${NODE}" "${OVNKUBE_NODE_POD}" "ovn-controller" \
+      > ${NETWORK_TRACE_PATH}/${NODE}_${OVNKUBE_NODE_POD}_ovn-controller.log &
+    PIDS+=($!)
+
+    oc -n openshift-ovn-kubernetes logs ${OVNKUBE_NODE_POD} -c ovn-controller -p 2>/dev/null | \
+      convert_ovs_timestamp "${NODE}" "${OVNKUBE_NODE_POD}" "ovn-controller" \
+      > ${NETWORK_TRACE_PATH}/${NODE}_${OVNKUBE_NODE_POD}_ovn-controller-previous.log &
+    PIDS+=($!)
+
+    oc -n openshift-ovn-kubernetes logs ${OVNKUBE_NODE_POD} -c ovnkube-node 2>/dev/null | \
+      convert_log_timestamp "${NODE}" "${OVNKUBE_NODE_POD}" "ovnkube-node" \
+      > ${NETWORK_TRACE_PATH}/${NODE}_${OVNKUBE_NODE_POD}_ovnkube-node.log 2>/dev/null &
+    PIDS+=($!)
+
+    oc -n openshift-ovn-kubernetes logs ${OVNKUBE_NODE_POD} -c ovnkube-node -p 2>/dev/null | \
+      convert_log_timestamp "${NODE}" "${OVNKUBE_NODE_POD}" "ovnkube-node" \
+      > ${NETWORK_TRACE_PATH}/${NODE}_${OVNKUBE_NODE_POD}_ovnkube-node-previous.log  &
+    PIDS+=($!)
+
+    if [[ ${OVNKUBE_MASTER} != "" ]] ; then
+       # northd nbdb sbdb ovnkube-master
+      oc -n openshift-ovn-kubernetes logs ${OVNKUBE_MASTER} -c northd 2>/dev/null | \
+        convert_ovs_timestamp "${NODE}" "${OVNKUBE_MASTER}" "northd" \
+	> ${NETWORK_TRACE_PATH}/${NODE}_${OVNKUBE_MASTER}_northd.log &
+      PIDS+=($!)
+
+      oc -n openshift-ovn-kubernetes logs ${OVNKUBE_MASTER} -c nbdb 2>/dev/null | \
+        convert_ovs_timestamp "${NODE}" "${OVNKUBE_MASTER}" "nbdb" \
+        > ${NETWORK_TRACE_PATH}/${NODE}_${OVNKUBE_MASTER}_nbdb.log &
+      PIDS+=($!)
+
+      oc -n openshift-ovn-kubernetes logs ${OVNKUBE_MASTER} -c sbdb 2>/dev/null | \
+        convert_ovs_timestamp "${NODE}" "${OVNKUBE_MASTER}" "sbdb" \
+        > ${NETWORK_TRACE_PATH}/${NODE}_${OVNKUBE_MASTER}_sbdb.log &
+      PIDS+=($!)
+
+      oc -n openshift-ovn-kubernetes logs ${OVNKUBE_MASTER} -c ovnkube-master 2>/dev/null | \
+        convert_log_timestamp "${NODE}" "${OVNKUBE_MASTER}" "ovnkube-master" \
+        > ${NETWORK_TRACE_PATH}/${NODE}_${OVNKUBE_MASTER}_ovnkube-master.log &
+      PIDS+=($!)
+
+     fi
+
+   done
+}
+
+function convert_journal_timestamp {
+  sed 's/Jan /01/; s/Feb /02/; s/Mar /03/; s/Apr /04/; s/May /05/; s/Jun /06/;
+  s/Jul /07/; s/Aug /08/; s/Sep /09/; s/Oct /10/; s/Nov /11/; s/Dec /12/;
+  s/ /-/' | sed '/^---/d'
+}
+
+function gather_service_data {
```

Comment:
We already have scripts, in this repo that collect this why not simply call those?

Reply:
There is a post process step that converts the date to a common format.

Reply:
Should we do this in other places too? Should we generalize this?

Reply:
@sferich888 I think a common timestamp and prefix in all of our logging would be helpful. With that we can generate a variety of traces picking up interesting logs. Also, scripts can be written to analyze the traces for expected and deviant behavior. I was interested in finding out how ovn networking sets up a pod's network. The trace is very enlightening, it shows how processing goes from daemon to daemon. Pick a pod and grep for it in the trace.

The downside is log lines get longer and files get bigger.

SOAPBOX
In a cluster there are numerous possible race conditions. For OCP to make it as a world class product we need to find and fix the races. Its a big part of making things "just work". In my years working on kernels, traces became the great salvation. Having a trace, like this, is the first small step. The heavy lifting comes in generating lines at important times and places in a fashion that is easy to analyze. That is the next step.
SOAPBOX OFF

This PR specifically generates a trace for ovn networking. I didn't want to expand and work with other teams until I got comfortable with doing this and analyzing results. I want to have something to show the other teams. My driving motivation is finding a way that anyone can look at a misbehaving cluster and quickly determine whether or not it is likely a network problem.